diff --git a/xwayland.c b/xwayland.c
index 3df7a08..9262e46 100644
--- a/xwayland.c
+++ b/xwayland.c
@@ -141,6 +141,8 @@ handle_xwayland_surface_destroy(struct wl_listener *listener, void *data)
 
 	wl_list_remove(&xwayland_view->destroy.link);
 	wl_list_remove(&xwayland_view->request_fullscreen.link);
+	wl_list_remove(&xwayland_view->associate.link);
+	wl_list_remove(&xwayland_view->dissociate.link);
 	xwayland_view->xwayland_surface = NULL;
 
 	view_destroy(view);
